File: Solutions/Medium/2536. Increment Submatrices by One/Solution.py
class Solution:
    def rangeAddQueries(self, n: int, queries: List[List[int]]) -> List[List[int]]:
        matrix_result: List[List[int]] = [[0 for _ in range(n)] for _ in range(n)]
        for query in queries:
            for i in range(query[0], query[2] + 1):
                matrix_result[i][query[1]] += 1
                if(query[3] < n - 1):
                    matrix_result[i][query[3] + 1] -= 1
        for row in matrix_result:
            for i in range(1, len(row)):
                row[i] += row[i - 1]
        return matrix_result

File: Solutions/Medium/2536. Increment Submatrices by One/Solution.cs
public class Solution {
    public int[][] RangeAddQueries(int n, int[][] queries) {
        int[][] matrixResult = new int[n][];
        for(int i = 0; i < n; i++){
            matrixResult[i] = new int[n];
        }
        foreach(int[] query in queries){
            for(int i = query[0]; i <= query[2]; i++){
                matrixResult[i][query[1]]++;
                if(query[3] < n - 1) matrixResult[i][query[3] + 1]--;
            }
        }
        foreach(int[] row in matrixResult){
            for(int i = 1; i < n; i++){
                row[i] += row[i - 1];
            }
        }
        return matrixResult;
    }
}